app/controllers/users_controller.rb

```diff
@@ -165,19 +165,21 @@ def remove_identity
   end
 
   def onboarding_update
+    authorize User
+    current_user.saw_onboarding = true
+
     if params[:user]
       sanitize_user_params
       current_user.assign_attributes(params[:user].permit(ALLOWED_USER_PARAMS))
       current_user.profile_updated_at = Time.current
     end
 
-    if current_user.save && params[:profile]
+    if current_user.invalid?
```

Comment:
I know you're primarily updating code, but the `Profiles::Update` service is taking care of quite a few things that also happen here, like updating the `profile_updated_at` timestamp, etc. It also runs user and profile updates in a transaction, so it will either update both or neither.

It may be worth investigating if you could reduce this method to generating the correct user attributes and passing them into the service together with `profile_params`. IMHO the more we can standardize behavior the better.

Reply:
That's a great suggestion. I believe this works as expected with only a minor tweak.